python_modules/dagster/dagster/_core/storage/db_io_manager.py

```diff
@@ -26,8 +27,9 @@
 
 
 class TablePartition(NamedTuple):
-    time_window: TimeWindow
     partition_expr: str
+    time_window: Optional[TimeWindow]
```

Comment:
I would lean towards combining these into a single `partition` field that's `Union[TimeWindow, str]`, because exactly one of them must be set.

Reply:
ok - i think it would still need a boolean or something to indicate if it's time or static partitioning so that we can make the correct select statement (== vs between). but then both of those would be required, which would be good

Reply:
isinstance(str) would also be fine